### PR TITLE
Added parameter "update_inverses"

### PR DESCRIPTION
**Pull request by sfetzel**
Using the parameter "update_inverses", you can reuse the inverse covariance matrices from a previous iteration. Example:
```
kfac = torch_kfac.KFAC(model, learning_rate=1e-3, damping=1e-3, update_cov_manually=True)
...
for epoch in range(100):
  model.zero_grad()
  with kfac.track_forward():
    # forward pass
    loss = ...
  with kfac.track_backward():
    # backward pass
    loss.backward()
  update_step = epoch % 10 == 0
  if update_step:
     kfac.update_cov()
  kfac.step(update_inverses=update_step)
  epoch += 1
...
```

### CODE REVIEW

**Comment by n-gao** [comment]
Thanks again for your PR and interest :) How is this different than the `update_cov_manually` parameter already present for the optimizer? It already allows you to manually decide when to update.

**Comment by sfetzel** [comment]
As far as I could see, the inverses of the covariance matrices are calculated in every step in the class `FisherBlock` in the method `multiply_preconditioner`, no matter if the covariance matrices are updated or not. As this is quite expensive in terms of operations, it would be great if the inverse matrices could be cached. This is what I implemented in the PR.

**Comment by n-gao** [comment]
I see, in that case couldn't we just always cache the inverses?

**Comment by sfetzel** [comment]
Yes we could also calculate the inverses in the `update_cov` function. I think that would be preferable. Do you agree on that?

**Comment by n-gao** [comment]
Yes that seems like the more elegant solution as it also avoids an additional parameter.
Thank you! :)

**Comment by sfetzel** [comment]
I though about this once more and the current implementation with the `update_inverses` parameter gives you the possibility to update the covariance matrices (which is fast) and have a moving average of the covariance matrices but still reuse the old inverse covariance matrices.

Anyways, I will move the `update_inverses` parameters to the `update_cov` method and calculate the inverses there as you suggested. In case you think the procedure I explained in the first paragraph does not make sense, we can just remove the parameter and still have cached inverse covariance matrices.

**Comment by n-gao** [comment]
Thanks for the changes! I am not 100% sure whether the unit tests add a lot since they heavily depend on the implementation itself. So they only check for consistency not for correctness.

**Comment by sfetzel** [comment]
I reviewed the unit tests again and improved the code. In summary we have:
- one unit test in `test_kfac_optimizer.py` which checks, that the inverses are updated when calling `KFAC.update_cov(..)`. I see that this unit test is questionable but it's too important to let this remain untested in my opinion.
- a unit test in `test_fisher_block.py` which checks that the inverse covariance matrices are reused and not calculated again.
- a unit test in `test_fisher_block.py`  which checks the correctness of the inverse covariance matrices by multiplying a random gradient with the expected inverse covariance matrices.

Could you please check again and let me know if any unit test should be improved?
